ui/src/common/components/SearchBox/searchHistory.js

```diff
@@ -0,0 +1,42 @@
+import { SEARCH_BOX_NAMESPACES } from '../../../reducers/search';
+import LRASet from './LRASet';
+
+const STORAGE_KEY = 'search-box-history';
+const HISTORY_LIMIT = 25;
+
+function runWhenIdle(task) {
```

Comment:
idle js everywhere

Reply:
this is when js engine is idle, not the user so unfortunately we can't use `idlejs` 😭  😃